Beta v6.34.3

MichaIng opened the pull request:
### Beta v6.34.3
_2020-12-20_

#### Changes since v6.34.2
- DietPi-Config | RPi: When enabling I2C support, the Python 3 SMbus module is now installed instead of the deprecated Python 2 version. This matches as well the dependency of PiJuice, which nowadays pulls Python 3 SMbus as package dependency, hence this change avoids the obsolete install of the whole Python 2 stack when PiJuice is installed through dietpi-software. Many thanks to @Trunkzeh for reporting this issue: https://github.com/MichaIng/DietPi/issues/3963#issuecomment-748547394

As no big changes are done with this RC, it will me merged together with the v6.34 release.

MichaIng committed:
v6.34

+ DietPi-Survey | The exit code was assigned wrong when purging data succeeded. It should have been "2" according to code comment, but was "1", matching the case of upload failure. Since we do not parse the exit code anywhere and a non-zero code IMO should mean an error, we'll now simply pass the curl exit code. Whether data has been uploaded or purged can be derived from the SURVEY_OPTED_IN dietpi.txt setting, if required.

MichaIng committed:
v6.34

+ DietPi-Software | OctoPrint: Do not use pip cache at all, which matches the way OctoPrint itself invokes pip and saves ~30 MiB disk space
+ DietPi-Software | OctoPrint: Remove sudoers config on uninstall

MichaIng committed:
v6.34 (#3983)

+ DietPi-Software | Desktops:  Replace DietPi desktop icon with transparent one
+ DietPi-Software | Pull desktop icon from website repository which is the central host for all sizes of DietPi icons now, but leave the icon on this repo for a while, for older DietPi versions

MichaIng committed:
v6.34

+ README | Add user profile links for previous contributors (Hall of Fame) as well

MichaIng committed:
v6.34

+ DietPi-Banner | Remove left apostrophe from donations link

MichaIng committed:
v6.34

+ DietPi-Software | PHP: Replace 7 years old opcache.php with a modern replacement that for some strange reason matches our sources list in credits already

fpetru commented:
@MichaIng - Viewing the latest changes, I will update the release notes & web tools with the new opcache gui - https://github.com/amnuts/opcache-gui.

MichaIng commented:
Practically nothing really changes. This OPcache status page simply has a refreshed layout and uses modern browser technologies, but contains the exact same information, no need to promote this IMO 😉.

MichaIng committed:
v6.34

+ DietPi-Set_software | Install Python 3 SMBus module instead of deprecated Python 2 one

MichaIng committed:
v6.34

+ DietPi-Software | PiJuice: Since I2C has a software ID, pull it transparently via pre-requirements loop
+ DietPi-Software | PiJuice: i2c-tools and python3(!)-smbus are pulled as package dependencies and installed together with I2C, hence no need to pull them again explicitly
+ DietPi-Software | PiJuice: "AudioPhonics" => "Audiophonics"

MichaIng committed:
v6.34

+ CHANGELOG | DietPi-Config on RPi: When enabling I2C support, the Python 3 SMbus module is now installed instead of the deprecated Python 2 version

MichaIng committed:
v6.34

+ RC up

MichaIng committed:
v6.34

+ DietPi-Globals | Default RC up

MichaIng committed:
v6.34

+ DietPi-Set_hardware | soundcard: Error out when no sound card string ($2) has been passed

MichaIng committed:
v6.34

+ DietPi-Globals | G_WHIP_MENU: Stay in menu if empty option was selected (separator line). Exit loop in case of cancel button selection or error or if .WHIP_MENU_RESULT could not be created.

MichaIng committed:
v6.34

+ DietPi-Software | Nextcloud: Apply 512 MiB PHP memory limit which is required to assure updater success and will else be treated as error with Nextcloud 21: https://github.com/nextcloud/server/commit/1995097cb8623b2447a2805a8e8e045c24203311

MichaIng committed:
v6.34

+ DietPi-Patch | Nextcloud: Apply 512 MiB PHP memory limit which is required to assure updater success and will else be treated as error with Nextcloud 21: https://github.com/nextcloud/server/commit/1995097cb8623b2447a2805a8e8e045c24203311
